src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +209,37 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
```

Comment:
I think it's enough to tell the user that it's not possible. The "Please stop[...]" sounds like a help message for interactive programs. ;) (Like: Now please press the "Stop" button and proceeed)